src/aws-cpp-sdk-transfer/source/transfer/TransferManager.cpp

```diff
@@ -1419,5 +1433,53 @@ namespace Aws
             auto handle = CreateUploadFileHandle(fileStream.get(), bucketName, keyName, contentType, metadata, context, fileName);
             return SubmitUpload(handle);
         }
+
+        Aws::Utils::ByteBuffer TransferManager::GetChecksumForAlgorithm(Aws::S3::Model::ChecksumAlgorithm algorithm,
```

Comment:
is this really needed? if tm sets algo on UploadPart, shouldnt it force calculation of checksum during execution of that api and then you can retrieve calculated checksum? This will result in checksum being calculated twice

Reply:
Yes we do because when we call `CompleteMultipartUpload` we need to provide it the checksum of each part uploaded. so we have to keep track of it. as far as it being calculated twice, yes that is unfortunate and we could likely refactor it to a virtual function and do a conditional calculation, but that can be a nice to have we can do later

Reply:
nope theres a easy around this, implementing